content/doc/developer/internationalization/internationalization-and-localization.adoc

```diff
@@ -0,0 +1,301 @@
+you might consider using ’ - the unicode character U+2019, written in
+properties file as \u2019.
+|===
+
+=== Marking messages in jelly files
+
+Your jelly files in `+src/main/resources+` often contain messages that
+need to be localized, and those need to be marked as well.
+
+In the simplest case, suppose you have a part of a Jelly file that looks
+like the following:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<h1>Output</h1>
+<p>...</p>
+....
+
+|===
+
+Change this to the following:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<h1>${%Output}</h1>
+<p>...</p>
+....
+
+|===
+
+The $\{%...} marker tells stapler,
+(explained in link:doc/developer/architecture/web/) to look for localized resources,
+and, if none are found, it prints "Output" for this, which is what
+you want.
+
+Let's consider the case where the localization requires parameters.
+Suppose you have a Jelly file `+foo.jelly+` like this:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<p>Click <a href="${obj.someMethod(a,b,c)}" >here</a></p>
+....
+
+|===
+
+In this case, first you must write `+foo.properties+` for the default
+locale:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+click.here.link=Click <a href="{0}" >here</a>
+....
+
+|===
+
+Then update `+foo.jelly+` to refer to this:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<p>${%click.here.link(obj.someMethod(a,b,c))}</p>
+....
+
+|===
+
+You can pass multiple parameters, separating them with ','
+(just like a function call), and from property files you can reference
+them as \{0}, \{1}, etc., by following
+https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/MessageFormat.html[the
+MessageFormat class convention].
+
+Now consider the case where you put a message in an expression.
+Suppose you have a Jelly file like this:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<p>${h.ifThenElse(x,"no error","error")}</p>
+....
+
+|===
+
+You can mark those two strings for localization like this:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<p>${h.ifThenElse(x,"%no error","%error")}</p>
+....
+
+|===
+
+Finally, because you can use expressions in Jelly, you need to ensure
+your regular strings don't contain brackets. +
+For example, this will *not* parse correctly, as it will be parsed as an
+expression:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<h1>${%Path to file (.ipa or .apk)}</h1>
+....
+
+|===
+
+You need to create two separate strings in this case:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<h1>${%Path to file} (${%.ipa or .apk})</h1>
+....
+
+|===
+
+=== Create of html files for help tips.
+
+You can configure entries in jelly files that display a tip message when the user clicks on a help icon.
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<f:entry title="${%title}" help="/plugin/myPlugin/help.html">
+....
+
+|===
+
+Then, just create the file help.html for English or help_xx.html for the
+'xx' language.
+
+== What translators need to know/do
```

Comment:
Simplified phrasing of the heading (optional)
```suggestion
== Translator tasks
```

Reply:
Done.